syntax = "proto3";

package auth;

message User {
  uint64 id = 1;
  string username = 2;
}

message Token {
  string token = 1;
}

message Message {
  string message = 1;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  Token token = 1;
}

message SignupRequest {
  string username = 1;
  string password = 2;
}

message ValidateTokenRequest {
  string token = 1;
}

message ValidateTokenResponse {
  User user = 1;
  Message message = 2;
}

service AuthService {
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc Signup(SignupRequest) returns (LoginResponse);
  rpc ValidateToken(ValidateTokenRequest) returns (ValidateTokenResponse);
}